tests/unit/amo/pages/TestAddon-new.js

```diff
@@ -94,79 +137,1135 @@ describe(__filename, () => {
+    // to the 451 error.
+    expect(
+      screen.getByText('That page is not available in your region'),
+    ).toBeInTheDocument();
+  });
+
+  it('passes props to the Page component', () => {
+    renderWithAddon();
+
+    // By passing isAddonInstallPage to Page, the GetFirefoxBanner is not shown.
+    expect(screen.queryByText('download Firefox')).not.toBeInTheDocument();
+
+    // The WrongPlatformWarning is shown, but it is the one embedded by Addon,
+    // not the one from Page.
+    expect(
+      screen.getByClassName('Addon-WrongPlatformWarning'),
+    ).toBeInTheDocument();
+    expect(
+      screen.queryByClassName('Page-WrongPlatformWarning'),
+    ).not.toBeInTheDocument();
+  });
+
+  describe('VPNPromoBanner integration', () => {
+    beforeEach(() => {
+      const fakeConfig = getMockConfig({
+        enableFeatureVPNPromo: true,
+        experiments: {
+          [EXPERIMENT_CONFIG.id]: true,
+        },
+      });
+      config.get.mockImplementation((key) => {
+        return fakeConfig[key];
+      });
+
+      // Write a cookie that will allow the Banner to appear.
+      createVPNExperimentCookie(VARIANT_SHOW);
+    });
+
+    it('passes showVPNPromo as `true` to Page if the add-on is an extension', () => {
+      renderWithAddon();
+
+      expect(
+        screen.getByRole('link', { name: 'Get Mozilla VPN' }),
+      ).toBeInTheDocument();
+    });
+
+    it.each([ADDON_TYPE_DICT, ADDON_TYPE_LANG, ADDON_TYPE_STATIC_THEME])(
+      'passes showVPNPromo as `false` to Page if the add-on is a %s',
+      (type) => {
+        addon.type = type;
+        renderWithAddon();
+
+        expect(
+          screen.queryByRole('link', { name: 'Get Mozilla VPN' }),
+        ).not.toBeInTheDocument();
+      },
+    );
+
+    it('passes showVPNPromo as `false` to Page if the add-on is null', () => {
+      render();
+
+      expect(
+        screen.queryByRole('link', { name: 'Get Mozilla VPN' }),
+      ).not.toBeInTheDocument();
+    });
+  });
+
+  const testServerRedirect = ({ slugInURL }) => {
+    const location = `/${lang}/${clientApp}/addon/${slugInURL}/`;
+    store.dispatch(loadAddon({ addon, slug: slugInURL }));
+    const dispatch = jest.spyOn(store, 'dispatch');
+    renderWithAddon({ location });
+
+    expect(dispatch).toHaveBeenCalledWith(
+      sendServerRedirect({
+        status: 301,
+        url: getLocation(addon.slug),
+      }),
+    );
+    return dispatch;
+  };
+
+  it('dispatches a server redirect when slug is a numeric ID', () => {
+    const dispatch = testServerRedirect({ slugInURL: addon.id });
+    // 1. Initial LOAD_ADDON
+    // 2. @@router/LOCATION_CHANGE, which happens on every page load
+    // 3. SEND_SERVER_REDIRECT
+    // 4. FETCH_CATEGORIES (initiated by AddonMoreInfo)
+    // 5. FETCH_ADDONS_BY_AUTHORS (initiated by AddonsByAuthorsCard)
+    // 6. FETCH_RECOMMENDATIONS (initiated by AddonRecommendations)
+
+    expect(dispatch).toHaveBeenCalledTimes(6);
```

Comment:
are these individual dispatches tested in other tests?  If not, can we test them here?

Reply:
They are all tested elsewhere, but considering that we have these comments here, and are counting the calls, I guess it makes sense to also assert about them. My only concern would be that it might require a bunch of set up to know exactly how they are being called in the test, but if that is the case I could just use the basic `expect.objectContaining({ type: THE_TYPE })` syntax. I will add this.